framework/include/base/Assembly.h

```diff
@@ -258,6 +258,12 @@ class Assembly
    */
   const MooseArray<Point> & normals() const { return _current_normals; }
 
+  /**
+   * Returns the array of tangents for quadrature points on a current side
+   * @return A _reference_.  Make sure to store this as a reference!
+   */
+  const MooseArray<std::vector<Point>> & tangents() const { return _current_tangents; }
```

Comment:
Are these guaranteed to be an orthogonal basis of the tangent space?

Reply:
Yes. These actually come out of libMesh